### DISCUSSION TITLE
Even after latest Update, when feeding Midi to Tsynth, Heavy Lag/Unresponsiveness

### DISCUSSION THREAD

**Discussion by velvetwafer** [ElectroTechnique/electrotechnique.github.io#80] [Software Issues]
After a while of using the Tsynth, i started to realize, that whenever the Thing gets Note Sequences fed by Midi, certain Functions sometimes completely hang, like Cutoff, Resonance, FX Mix....Teensy wont see any Inputs to them,and acts as if i hadnt turned that Potentiometer.. only sometimes after several tries they start to function again. As soon as i stop the Sequencing, and only play single Notes via Hand, everything is back to normal....makes this Synth difficult to be properly used in a live setting...you can dial in the Settings before playing, and everything is fine...but dont dare turning the wrong Pot (Some Pots like Oscillator Tune seems to be exempt from this, as is Noise /LFO Level). Thanks for your Help in Advance!

**Comment by Electrotechnique** [comment]
What sequencer are you using, how many notes per second and how long is it before it starts to lag? It's probably running out of resources.

**Comment by velvetwafer** [comment]
Im using a Synthstrom Deluge, and i think was feeding it a single Note per Step, 16 Steps long...it instantly lags as soon as i start the Sequencer on the Deluge. Not much, but its easily reproducable...i think it has a Teensy 3.6 in it, these were the ones originally recommended

**Reply by Electrotechnique** [reply]
If you can compile the code yourself, uncomment the following and try again. Perhaps we just need a bigger MIDI buffer:

Try this:

`//MIDIDevice midi1(myusb);`
` MIDIDevice_BigBuffer midi1(myusb); // Try this if your MIDI Compliant controller has problems`
